### DISCUSSION TITLE
How is polling rate decided in host mode?

### DISCUSSION THREAD

**Discussion by JonnyHaystack** [hathach/tinyusb#1654] [Q&A] [answered]
Hi, I've been wondering, how exactly is the polling rate determined when operating in host mode? Is bInterval automatically taken into account? Or does it default to 1000Hz? Or do you have to manually limit it in user code? I can't see where in the code this would be controlled.

**Comment by hathach** [comment]
bInterval is declared by device, Ideally, it should be the one be used. However depending the host driver implementation (for a simpler code, driver can ignore or respect this). What is your mcu ?

**Reply by JonnyHaystack** [reply]
RP2040. I couldn't find in the code where it would be looking at the bInterval from the device and deciding polling rate based on that. I would also like to be able to manually set a higher polling rate if needed (because some devices set a stupid low polling rate for no reason).

**Comment by hathach** [comment, marked as answer]
for rp2040, polling is already supported by hw controller and correctly set by hcd driver 

https://github.com/hathach/tinyusb/blob/master/src/portable/raspberrypi/rp2040/hcd_rp2040.c#L334

![image](https://user-images.githubusercontent.com/249515/194874646-ecdd8a20-b28c-43fa-afb7-bc4ce22a4f58.png)

**Reply by JonnyHaystack** [reply]
I see. Thanks for the explanation. Looks like there's no way to override this then.
